modules/transactions/src/main/java/org/apache/ignite/internal/tx/impl/TxManagerImpl.java

```diff
@@ -629,14 +636,22 @@ public CompletableFuture<Void> start() {
 
         messagingService.addMessageHandler(ReplicaMessageGroup.class, this);
 
+        messagingService.addMessageHandler(TxMessageGroup.class, (message, senderConsistentId, correlationId) -> {
+            rwTxFinishedHandlerLock.writeLock().lock();
```

Comment:
It looks like we can move `rwTxFinishedHandlerLock`, handler subscription, this method and `newBeginTimestamp()` back to `RwTimestampFinishedHandler`; it seems natural to have it there and not clutter `TxManagerImpl`.

Also, the write lock now covers too much, even sending response is included. It seems that it's enough to only do `isRwTransactionsFinished()` under the lock.

Reply:
fix it